Formula/julia.rb

```diff
@@ -79,6 +79,8 @@ class Julia < Formula
     depends_on "libunwind"
   end
 
+  conflicts_with "juliaup", because: "both install `julia` binaries"
```

Comment:
Let's add the `conflicts_with` to both formulae in a separate PR so we don't have to rebuild `julia`.

Reply:
Done: https://github.com/Homebrew/homebrew-core/pull/93496